src/core/gamemodes.h

```diff
@@ -0,0 +1,23 @@
+#ifndef GAMEMODES_H
```

Comment:
There is a generator in classgenerator -g "enum". It currently uses AutoId-s instead of string. You should use that enum<>string for conversion. That is the current standard.